pkg/tenant/projects.go

```diff
@@ -522,6 +568,155 @@ func (s *Server) ProjectUpdate(c *gin.Context) {
+	}
+
+	// Parse the projectID from the URL
+	projectParam := c.Param("projectID")
+	if projectID, err = ulid.Parse(projectParam); err != nil {
+		sentry.Warn(c).Err(err).Str("projectID", projectParam).Msg("could not parse project id from URL")
+		c.JSON(http.StatusNotFound, api.ErrorResponse("project not found"))
+		return
+	}
+
+	// Patch the project, this method handles the rest of the responses and errors.
+	s.projectPatch(c, orgID, projectID)
+}
+
+// projectPatch is a handler for multiple patch project routes that applies a partial
+// update to a project. Only the provided fields are updated, and an error is returned
+// if any field does not exist or if no updates were made.
+func (s *Server) projectPatch(c *gin.Context, orgID, projectID ulid.ULID) {
+	var (
+		err     error
+		project *db.Project
+		patch   map[string]interface{}
+	)
+
+	// Parse the patch fields from the request body
+	if err = c.BindJSON(&patch); err != nil {
+		sentry.Warn(c).Err(err).Msg("could not parse project patch request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(api.ErrUnparsable))
+		return
+	}
+
+	if len(patch) == 0 {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("no fields provided for patch"))
+		return
+	}
+
+	// Verify that the project exists in the user's organization.
+	if err = db.VerifyOrg(c, orgID, projectID); err != nil {
+		if errors.Is(err, db.ErrNotFound) || errors.Is(err, db.ErrOrgNotVerified) {
+			c.JSON(http.StatusNotFound, api.ErrorResponse("project not found"))
+			return
+		}
+		sentry.Warn(c).Err(err).Msg("could not check verification")
+		c.JSON(http.StatusUnauthorized, api.ErrorResponse("could not verify organization"))
+		return
+	}
+
+	// Retrieve the project from the database.
+	if project, err = db.RetrieveProject(c, projectID); err != nil {
+		if errors.Is(err, db.ErrNotFound) {
+			c.JSON(http.StatusNotFound, api.ErrorResponse("project not found"))
+			return
+		}
+		sentry.Error(c).Err(err).Msg("could not retrieve project")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not lookup project"))
+		return
+	}
+
+	// Iterate over the patch fields and perform the updates.
+	for field, value := range patch {
+		var ok bool
+		switch field {
+		case "name":
+			var name string
+			if name, ok = value.(string); !ok {
+				c.JSON(http.StatusBadRequest, api.ErrorResponse(api.FieldTypeError(field, "string")))
+				return
+			}
+
+			if name == project.Name {
+				c.JSON(http.StatusBadRequest, api.ErrorResponse("project already has this name"))
+				return
+			}
+
+			project.Name = name
+		case "description":
+			var description string
+			if description, ok = value.(string); !ok {
+				c.JSON(http.StatusBadRequest, api.ErrorResponse(api.FieldTypeError(field, "string")))
+				return
+			}
+
+			if description == project.Description {
+				c.JSON(http.StatusBadRequest, api.ErrorResponse("project already has this description"))
+				return
+			}
+
+			project.Description = description
+		case "owner":
```

Comment:
Should this be `owner_id`? I don't like having to rely on hardcoded field names, is there something in the `confire` library that would help us parse the fields more elegantly?

Reply:
It should match the field in the `Project` struct - so it's owner, but then you're expecting the value to be a `map[string]interface{}` that has the `id` field inside of it (e.g. the serialized `Member` struct`). 

Confire does have tools that will make `PATCH` easier, so we should create a follow on story to update it.